object/slicer/slicer.go

```diff
@@ -164,20 +163,44 @@ func slice(ctx context.Context, ow ObjectWriter, header object.Object, data io.R
 	objectPayloadLimit := childPayloadSizeLimit(opts)
 
 	var n int
-	bChunk := opts.payloadBuffer
-	if bChunk == nil {
-		bChunk = make([]byte, objectPayloadLimit)
-	}
 
 	writer, err := initPayloadStream(ctx, ow, header, signer, opts)
 	if err != nil {
 		return rootID, fmt.Errorf("init writter: %w", err)
 	}
 
+	var buf []byte
+	var buffered uint64
+
 	for {
-		n, err = data.Read(bChunk)
+		buffered = writer.rootMeta.length
```

Comment:
why do we check these numbers and not check the corresponding slices' sizes?

Reply:
> corresponding slices' sizes

u mean `len`? They have static length